File: LeetCode/LeetCode Practice Problems/1021- Remove Outermost Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        string s = "" , ans = "";
        int c = 0;
        for(int i=0;i<S.length();i++)
        {
            if(S[i] == '(')
            {
                c++;
                if(c>1)
                    ans = ans + "(";
            }
            else
            {
                c--;
                if(c>0)
                    ans = ans + ")";
            }
            
            if(c == 0)
            {
                s = s + ans;
                ans = "";
            }
        }
        
        return s;
    }
};

File: LeetCode/LeetCode Practice Problems/1021-Remove Outermost Parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        String ans = "";
        char ch[] = s.toCharArray();
        int n = ch.length;
        
        for(int i=0;i<n;) {
            if(ch[i] == '(') {
                int cnt = 1;
                i++;
                while(i<n) {
                    if(ch[i] == ')' && cnt == 1) {
                        break;
                    }
                    if(ch[i] == '(') {
                        cnt++;
                    } else {
                        cnt--;
                    }
                    ans = ans + ch[i];
                    i++;
                }
                i++;
                
            } else {
                i++;
            }
        }
        
        return ans;
    }
}
